src/main/java/jeeves/parser/Parser.java

```diff
@@ -0,0 +1,221 @@
+            // Checks if the user has entered the command properly
+            // and extracts the relevant information to parse
+            // then returns args <command, desc, fromDate, toDate>
+            try {
+                int fromDateIndex = getFromDateIndex(input);
+                int toDateIndex = getToDateIndex(input, fromDateIndex);
+                String desc = input.substring(FINDCOMMAND_EVENT_OFFSET, fromDateIndex - 1);
+                String fromDate = input.substring(fromDateIndex + FINDFIELD_FROM_OFFSET, toDateIndex - 1);
+                String toDate = input.substring(toDateIndex + FINDFIELD_TO_OFFSET);
+                args.add(desc);
+                args.add(fromDate);
+                args.add(toDate);
+            } catch (MissingDescriptionException | MissingFromException | MissingToException e) {
+                System.out.println(e.getMessage());
+            }
+        } else if (input.startsWith("find ")) {
+            // Gets the search term the user wishes to find
+            // and returns args <command, findTerm>
+            args.add("find");
+            String findTerm = input.substring(FINDCOMMAND_FIND_OFFSET);
+            args.add(findTerm);
+        } else if (input.equals("bye")) {
+            args.add(input);
+        }
+        return args;
+    }
+
+    /**
+     * Checks if a string is only made up of integer numbers.
+     * Uses regex to determine if a string contains only integers
+     *
+     * @param input The string to be checked
+     * @return Whether the string is fully comprised of integers
+     */
+    static boolean isNotNumber(String input) {
+        return !input.matches("[0-9]+");
+    }
+
+    /**
+     * Checks if the string provided can be parsed into a valid id Integer.
+     * Throws a custom exception with an appropriate error message upon encountering an error.
+     * Currently, checks if the idString is: Present, is a Number, is within the usable range.
+     *
+     * @param idString The string to be checked.
+     */
+    static void checkIfValidId(String idString) {
+        try {
+            if (idString.isEmpty()) {
+                // id field is empty
+                throw new MissingIdException();
+            } else if (isNotNumber(idString)) {
+                // id field is not an integer
+                throw new NotIntegerIdException();
+            } else if (Integer.parseInt(idString) > Task.getTaskCount()) {
+                // id does not exist
+                throw new OutOfBoundIdException();
+            } 
+        } catch (MissingIdException | NotIntegerIdException | OutOfBoundIdException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Checks if the string provided is not empty
+     * Throws a custom exception with an appropriate error message if it is.
+     * Else does nothing.
+     *
+     * @param desc The string to be checked.
+     */
+    static void checkIfDescMissing(String desc) {
```

Comment:
is this better `isDescMissing()`

Reply:
Attempting to avoid confusing method names as boolean methods typically use the is....() format and this method is void, thanks for the review